artifactory/commands/container/dockerfileutils/parser.go

```diff
@@ -0,0 +1,145 @@
+package dockerfileutils
+
+import (
+	"bufio"
+	"github.com/jfrog/jfrog-cli-artifactory/artifactory/commands/ocicontainer"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/jfrog/jfrog-client-go/utils/log"
+)
+
+// ParseDockerfileBaseImages extracts all base image references from FROM instructions in a Dockerfile
+// Handles FROM instructions with flags like --platform and AS clauses
+// Ignores FROM clauses that reference previous build stages in multi-stage builds
+// Returns base images along with OS and architecture extracted from --platform flag or runtime defaults
+// Examples:
+//   - FROM ubuntu:20.04
+//   - FROM ubuntu:20.04 AS builder
+//   - FROM --platform=linux/amd64 ubuntu:20.04
+//   - FROM --platform=linux/amd64 ubuntu:20.04 AS builder
+//   - FROM builder (skipped - references previous stage)
+func ParseDockerfileBaseImages(dockerfilePath string) ([]ocicontainer.BaseImage, error) {
+	file, err := os.Open(dockerfilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if cerr := file.Close(); cerr != nil {
```

Comment:
```suggestion
		if closeErr := file.Close(); closeErr != nil {
```

what is cerr? It will be much easier to read if expanded